packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -0,0 +1,820 @@
+    let splitAddress
+    if (createSplitArgs.salt) {
+      splitAddress = await factory.read.predictDeterministicAddress([
+        {
+          recipients: createSplitArgs.recipients,
+          allocations: createSplitArgs.allocations,
+          totalAllocation: totalAllocation,
+          distributionIncentive: createSplitArgs.distributionIncentive,
+        },
+        createSplitArgs.owner,
+        createSplitArgs.salt,
+      ])
+    } else {
+      splitAddress = await factory.read.predictDeterministicAddress([
+        {
+          recipients: createSplitArgs.recipients,
+          allocations: createSplitArgs.allocations,
+          totalAllocation: totalAllocation,
+          distributionIncentive: createSplitArgs.distributionIncentive,
+        },
+        createSplitArgs.owner,
+      ])
+    }
+
+    return {
+      splitAddress,
+    }
+  }
+
+  async isDeployed(createSplitArgs: CreateSplitV2Config): Promise<{
+    splitAddress: Address
+    deployed: boolean
+  }> {
+    validateAddress(createSplitArgs.owner)
+    createSplitArgs.recipients.map((recipient) => validateAddress(recipient))
+    this._requirePublicClient()
+
+    const factory = this._getSplitV2FactoryContract(createSplitArgs.splitType)
+
+    const totalAllocation = createSplitArgs.allocations.reduce((a, b) => a + b)
+
+    if (!createSplitArgs.salt) throw new Error('Salt required')
+    const [splitAddress, deployed] = await factory.read.isDeployed([
+      {
+        recipients: createSplitArgs.recipients,
+        allocations: createSplitArgs.allocations,
+        totalAllocation: totalAllocation,
+        distributionIncentive: createSplitArgs.distributionIncentive,
+      },
+      createSplitArgs.owner,
+      createSplitArgs.salt,
+    ])
+
+    return {
+      splitAddress,
+      deployed,
+    }
+  }
+
+  async getSplitBalance(
+    split: Address,
+    token: Address,
+  ): Promise<{
+    splitBalance: bigint
+    warehouseBalance: bigint
+  }> {
+    validateAddress(split)
+    this._requirePublicClient()
+
+    const splitContract = this._getSplitV2Contract(split)
+
+    const [splitBalance, warehouseBalance] =
+      await splitContract.read.getSplitBalance([token])
+
+    return {
+      splitBalance,
+      warehouseBalance,
+    }
+  }
+
+  async getReplaySafeHash(split: Address, hash: Hex): Promise<{ hash: Hex }> {
+    validateAddress(split)
+    this._requirePublicClient()
+
+    const splitContract = this._getSplitV2Contract(split)
+
+    const replaySafeHash = await splitContract.read.replaySafeHash([hash])
+
+    return {
+      hash: replaySafeHash,
+    }
+  }
+
+  async isValidSignature(
+    split: Address,
+    hash: Hex,
+    signature: Hex,
+  ): Promise<boolean> {
+    validateAddress(split)
+    this._requirePublicClient()
+
+    const splitContract = this._getSplitV2Contract(split)
+
+    return (
+      (await splitContract.read.isValidSignature([hash, signature])) ===
+      '0x1626ba7e'
```

Comment:
nit: what exactly is this? don't see it used anywhere else, maybe can still put it in a constants just so it's named/more clear?